src/qt/optionsmodel.cpp

```diff
@@ -70,9 +71,15 @@ void OptionsModel::Init(bool resetSettings)
     fMinimizeOnClose = settings.value("fMinimizeOnClose").toBool();
 
     // Display
-    if (!settings.contains("nDisplayUnit"))
-        settings.setValue("nDisplayUnit", BitcoinUnits::BTC);
```

Comment:
77edf1bd0149bd2625dff563f3bd4032a12a2233

Maybe it should migrate the old setting key if it exists?

Reply:
Will it be dead code after a while?

Reply:
And? For instance https://github.com/bitcoin/bitcoin/pull/20966 also tries to use old data.

Reply:
@promag 
I was thinking about your suggestion, and I failed to find a safe and short (about 1..2 lines of code) solution.
OTOH, adjusting this option requires only two clicks.

Therefore, I'm going to leave it as is, leaving a room for improvements in follow ups.